src/Gitcolab/Bundle/AppBundle/DependencyInjection/Compiler/GitUrlGeneratorPass.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * This file is part of Gitcolab.
+ *
+ * (c) Mbechezi mlanawo <mlanawo@mbechezi.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Gitcolab\Bundle\AppBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class GitUrlGeneratorPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $definition = $container->getDefinition('gitonomy_twig.url_generator');
+        $definition->setClass('Gitcolab\Bundle\AppBundle\Routing\GitUrlGenerator');
+
+        $definition->addMethodCall('setEntityManager', array($container->getDefinition('doctrine')));
```

Comment:
It still can since the good practice is to override the constructor. Even if you still use its parent